developersociety/commonslibrary
# Api tokens

For card: https://favro.com/organization/b03d6d6d9b11b61167ac4246/418c53a770389640d415a230?card=Dev-7463

Added simple token authentication using `rest_framework.authtoken`. Uses token to authenticate session, tokens are set on a per-user basis in the Django admin.

TO TEST:

(Test normal session auth)
- Create user and set token in django admin then logout.
- Once logged out, detail pages should be unavailable
   (e.g. `http://127.0.0.1:8000/api/v1/resources/71/`)
- Log in and revisit the same api view, full details should now show.

(Test token auth)
- Log out of all accounts
- CURL a detail page to make sure it is not available without token:
  (e.g. `curl 127.0.0.1:8000/api/v1/resources/71/`)
- CURL again with token as header to get access to view:
   (e.g. `curl -H "Authorization: Token <MY_TOKEN_HERE" 127.0.0.1:8000/api/v1/resources/71/`)

Taken inspiration from ACNY, if I've missed anything or if there is a more efficient way of going about it just let me know!
